ch_backup/ch_backup.py

```diff
@@ -104,6 +106,22 @@ def list(self, state: BackupState = None) -> Sequence[BackupMetadata]:
 
         return backups
 
+    def _lock_id(self, operation: str) -> Optional[str]:
+        lock_id = operation
+        # if the replica name part of the id starts with '$' it means
+        # we used `platform.node()`
+        missing_shard = "/UNKNOWN"
+        missing_replica = f"/${platform.node()}"
+        lock_id = operation
+        try:
+            shard, replica = self._context.ch_ctl.get_shard_and_replica_name()
+            lock_id += f"/{shard}" if shard else missing_shard
+            lock_id += f"/{replica}" if replica else missing_replica
+        except ClickhouseError:
+            logging.error("Error in forming lock_id.", exc_info=True)
+            lock_id = f"{lock_id}{missing_shard}{missing_replica}"
```

Comment:
Not sure that we need a fallback. It's better to choose one reliable approach for determining shard and replica names.

Reply:
Yes, that would be the best option. I think also that shard name could be neglected as knowing the replica (or hostname?) it is easy than to find the relative replica. What do you think guys? what would be the best approach? `platform.node()` only?

Reply:
Agree, shard name can be omitted.

As for reliable way for getting hostname I suggest to use `socket.getfqdn()`. We already use it, for example, in https://github.com/yandex/ch-tools/blob/25728d2e3287c2e19d9c14259a6f466ddfe38c3a/ch_tools/common/clickhouse/client/clickhouse_client.py#L25